### ISSUE TITLE
Potential Bug: Wrong file extension for Python file on macOS Catalina 10.15.4

### ISSUE THREAD

**Issue by christopherwolf1** [microsoft/vscode#98504]
Issue Type: <b>Bug</b>

1. Open up Visual Studio Code (and/or Insiders)
2. Create any new file
3. Attempt to save file as a "Python" file in the dropdown
(Animated GIF Below)

![Potential Bug](https://user-images.githubusercontent.com/15132959/82794618-9e00ff00-9e27-11ea-8660-ebade8b3bcae.gif)

I am pretty sure this is a bug; however I just checked on the Windows 10 version of VSC and while it doesn't assign *.gyp as an extension, instead using the proper *.py, it does list it as one of the extensions recognized as a Python file. 

As far as I am aware, I have never encountered *.gyp files in any coding context outside of node-gyp and associated infrastructure for Chromium.

VS Code version: Code 1.45.1 (5763d909d5f12fe19f215cbfdd29a91c0fa9208a, 2020-05-14T08:33:47.663Z)
OS version: Darwin x64 19.4.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.06GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
python|ms-|2020.5.80290
vscodeintellicode|Vis|1.2.7
vscode-icons|vsc|10.1.1

</details>

**Comment by kieferrm** [comment]
The following file extensions are registered as python: `".py", ".rpy", ".pyw", ".cpy", ".gyp", ".gypi", ".pyi", ".ipy"`. I don't know how the Save As dialog picks what extension to show by default.

**Comment by bpasero** [comment]
This seems like standard Electron behaviour, I can reproduce in a Fiddle and Electron seems to pick a random entry where I would expect it should pick the first:

```js
dialog.showSaveDialog(mainWindow, {
      filters: [
        {
          name: 'Python',
          extensions: [
            'py',  'rpy',
            'pyw', 'cpy',
            'gyp', 'gypi',
            'pyi', 'ipy'
          ]
        },
      ]
    });
```

**Comment by nogurenn** [comment]
Is there a workaround for this that we can set in our settings file or something?

**Comment by bpasero** [comment]
Reported as https://github.com/electron/electron/issues/25152

**Comment by bpasero** [comment]
This is fixed with Electron 9.3.0.

**Comment by bpasero** [comment]
\closedWith 89c002ab02f87102d91efc83c191ef1174756c6a
